cli/cmd/plugin/unmanaged-cluster/cluster/kind.go

```diff
@@ -158,6 +160,64 @@ func (kcm KindClusterManager) Prepare(c *config.UnmanagedClusterConfig) error {
 	return nil
 }
 
+// PreflightCheck performs and pre-checks that can find issues up front that
+// would cause problems for cluster creation.
+func (kcm KindClusterManager) PreflightCheck() []error {
+	// Check presence of docker
+	cmd := exec.Command("which", "docker")
+	if _, err := exec.Output(cmd); err != nil {
+		// In this case we can't check the rest of the settings, so just return
+		// the one error.
+		return []error{errors.New("docker CLI must be present")}
+	}
+
+	// Get Docker info
+	cmd = exec.Command("docker", "info")
+	output, err := exec.Output(cmd)
+	if err != nil {
+		return []error{fmt.Errorf("unable to get docker info: %w", err)}
+	}
+
+	issues := parseDockerInfoJSON(output)
+	if len(issues) == 0 {
+		return nil
+	}
+	return issues
```

Comment:
microscopic nit: consider inverting the if logic above so that this return statement returns nil. Makes it a little easier to read because you returns errors throughout the func and it'd be nice if the final return was nil (no error)

Reply:
Done